sorting/wiggle_sort.cpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * \addtogroup sorting Sorting Algorithms
+ * @{
+ * \file
+ * \brief [Wiggle Sort Algorithm]
+ * (https://khanhng.com/post/wiggle-sort/) Implementation
```

Comment:
I suggest using a standard recognized reference website rather than a personal one.
like this https://leetcode.com/problems/wiggle-sort-ii/, for example